target_chains/ethereum/contracts/contracts/aave/PythPriceOracleGetter.sol

```diff
@@ -47,19 +57,34 @@ contract PythPriceOracleGetter is PythAssetRegistry, IPriceOracleGetter {
             PythAssetRegistry.validTimePeriodSeconds()
         );
 
-        // price.price is always positive and therefore this cast is safe.
+        // Aave is not using any price feeds < 0 for now.
+        if (price.price <= 0) {
+            revert InvalidNonPositivePrice();
+        }
         uint256 normalizedPrice = uint64(price.price);
         // price.expo is bound between [-12, 12]
-        int32 normalizerExpo = price.expo + int8(BASE_EXPONENTS);
+        int32 normalizerExpo = price.expo + int8(BASE_NUM_DECIMALS);
+        bool isNormalizerExpoNeg = normalizerExpo < 0;
+        uint256 normalizer = isNormalizerExpoNeg
+            ? 10 ** uint32(-normalizerExpo)
+            : 10 ** uint32(normalizerExpo);
+
+        if (!isNormalizerExpoNeg && normalizer > type(uint192).max) {
+            revert NormalizationOverflow();
+        }
+
+        normalizedPrice = isNormalizerExpoNeg
+            ? normalizedPrice / normalizer
+            : normalizedPrice * normalizer;
 
-        if (normalizerExpo < 0) {
-            return normalizedPrice / (10 ** uint32(-normalizerExpo));
-        } else {
-            return normalizedPrice * (10 ** uint32(normalizerExpo));
+        if (normalizedPrice <= 0) {
+            revert InvalidNonPositivePrice();
         }
+
+        return normalizedPrice;
     }
 
-    function baseExponents(uint number) private pure returns (uint8) {
+    function baseNumExponents(uint number) private pure returns (uint8) {
```

Comment:
baseNumDecimals

I guess it would be nice to validate that the input is a power of 10 as well, though not required.

Reply:
added a check in the constructor to validate BASE_NUM_DECIMALS is a power of 10